pkg/statistics/handle/util/auto_analyze_proc_id_generator.go

```diff
@@ -14,27 +14,112 @@
 
 package util
 
+import (
+	"sync"
+
+	"github.com/pingcap/tidb/pkg/sessionctx/sysproctrack"
+	"golang.org/x/exp/maps"
+)
+
 // AutoAnalyzeProcIDGenerator is used to generate auto analyze proc ID.
 type AutoAnalyzeProcIDGenerator interface {
 	// AutoAnalyzeProcID generates an analyze ID.
 	AutoAnalyzeProcID() uint64
+	ReleaseAutoAnalyzeProcID(uint64)
 }
 
 var _ AutoAnalyzeProcIDGenerator = (*generator)(nil)
 
 type generator struct {
```

Comment:
Why is it a generator? Seems that the process is generated by the `domain.NextConnID`

Reply:
I think it just wants to isolate the dependency on the domain and provide a container to reference the application and release of the connect id.